lib/buffer.js

```diff
@@ -860,11 +860,11 @@ Buffer.prototype.equals = function equals(otherBuffer) {
 
   if (this === otherBuffer)
     return true;
-
-  if (this.byteLength !== otherBuffer.byteLength)
+  const len = this.byteLength;
```

Comment:
Could you try with `TypedArrayPrototypeGetByteLength` from primordials? I'd be curious to see if that would improve perf further or not.

Reply:
Good suggestion. I have tried `TypedArrayPrototypeGetByteLength`. As you can see in the updated benchmark result, it improved a lot.